upup/pkg/fi/cloudup/awstasks/external_target_group_attachment.go

```diff
@@ -0,0 +1,146 @@
+)
+
+//go:generate fitask -type=ExternalTargetGroupAttachment
+type ExternalTargetGroupAttachment struct {
+	Name      *string
+	Lifecycle *fi.Lifecycle
+
+	TargetGroupARN string
+
+	AutoscalingGroup *AutoscalingGroup
+}
+
+func (e *ExternalTargetGroupAttachment) name() string {
+	re := regexp.MustCompile("^.*:targetgroup/")
+	return re.ReplaceAllString(e.TargetGroupARN, "")
+}
+
+func (e *ExternalTargetGroupAttachment) Find(c *fi.Context) (*ExternalTargetGroupAttachment, error) {
+	cloud := c.Cloud.(awsup.AWSCloud)
+
+	if e.TargetGroupARN == "" {
+		return nil, fmt.Errorf("InstanceGroup did not have TargetGroupARNs set")
+	}
+
+	g, err := findAutoscalingGroup(cloud, *e.AutoscalingGroup.Name)
+	if err != nil {
+		return nil, err
+	}
+	if g == nil {
+		return nil, nil
+	}
+
+	for _, name := range g.TargetGroupARNs {
+		if aws.StringValue(name) != e.TargetGroupARN {
+			continue
+		}
+
+		actual := &ExternalTargetGroupAttachment{}
+		actual.TargetGroupARN = e.TargetGroupARN
+		actual.AutoscalingGroup = e.AutoscalingGroup
+
+		// Prevent spurious changes
+		actual.Name = e.Name // ELB attachments don't have tags
+		actual.Lifecycle = e.Lifecycle
+
+		return actual, nil
+	}
+
+	return nil, nil
+}
+
+func (e *ExternalTargetGroupAttachment) Run(c *fi.Context) error {
+	return fi.DefaultDeltaRunMethod(e, c)
+}
+
+func (s *ExternalTargetGroupAttachment) CheckChanges(a, e, changes *ExternalTargetGroupAttachment) error {
+	if a == nil {
+		if e.TargetGroupARN == "" {
+			return fi.RequiredField("TargetGroupARN")
+		}
+		if e.AutoscalingGroup == nil {
+			return fi.RequiredField("AutoscalingGroup")
+		}
+	}
+	return nil
+}
+
+func (_ *ExternalTargetGroupAttachment) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *ExternalTargetGroupAttachment) error {
+	if e.TargetGroupARN == "" {
+		return fi.RequiredField("TargetGroupARN")
+	}
+
+	request := &autoscaling.AttachLoadBalancerTargetGroupsInput{}
+	request.AutoScalingGroupName = e.AutoscalingGroup.Name
+	request.TargetGroupARNs = aws.StringSlice([]string{e.TargetGroupARN})
+
+	glog.V(2).Infof("Attaching autoscaling group %q to Target Group %q", fi.StringValue(e.AutoscalingGroup.Name), e.TargetGroupARN)
+	_, err := t.Cloud.Autoscaling().AttachLoadBalancerTargetGroups(request)
+	if err != nil {
+		return fmt.Errorf("error attaching autoscaling group to ELB: %v", err)
+	}
+
+	return nil
+}
+
+type terraformExternalTargetGroupAttachment struct {
+	TargetGroupARN   *terraform.Literal `json:"alb_target_group_arn,omitempty"`
+	AutoscalingGroup *terraform.Literal `json:"autoscaling_group_name,omitempty"`
+}
+
+func (_ *ExternalTargetGroupAttachment) RenderTerraform(t *terraform.TerraformTarget, a, e, changes *ExternalTargetGroupAttachment) error {
+	tf := &terraformExternalTargetGroupAttachment{
+		TargetGroupARN:   terraform.LiteralFromStringValue(e.TargetGroupARN),
+		AutoscalingGroup: e.AutoscalingGroup.TerraformLink(),
+	}
+
+	return t.RenderResource("aws_autoscaling_attachment", e.name(), tf)
```

Comment:
The response from `e.name()` is not unique when you have more than one IG attaching to the same ALB/NLB, we can probably use `e.name()+"-"+*e.AutoscalingGroup.Name` which is unique. The only reason I don't recomend using `*e.Name` like I do in `external_load_balancer_attachment.go` is because the ARN contains colon and terraform does not support them in the name.

Reply:
I've made `e.name()` now source from `*e.Name` and instead of strip the colon-containing parts of the arn only.